app/account/components/AccountDetails/components/AvatarUpdate/AvatarUpdate.tsx

```diff
@@ -0,0 +1,173 @@
+import { User } from "@interfaces"
+import { useUserStore } from "@stores"
+import { customFetch } from "@utils"
+import React, { useCallback, useState } from "react"
+import Cropper from "react-easy-crop"
+import "./avatarUpdate.scss"
+import { getCroppedImg } from "./cropUtils"
+
+const AvatarUpdate = () => {
+  const fetchUserData = useUserStore((state) => state.fetchUserData)
+  const [imageSrc, setImageSrc] = useState<string | null>(null)
+  const [crop, setCrop] = useState({ x: 0, y: 0 })
+  const [zoom, setZoom] = useState(1)
+  const [croppedAreaPixels, setCroppedAreaPixels] = useState<null | { width: number; height: number; x: number; y: number }>(null)
+  const [isCropping, setIsCropping] = useState(false)
+  const [userMsg, setUserMsg] = useState<string>()
+  const fileInputRef = React.useRef<HTMLInputElement | null>(null)
+  const user = useUserStore((state) => state.user)
+  const { username } = (user as User) ?? {}
+  const msgShowDuration = 2000
+
+  const closeModal = () => {
+    // reset cropper
+    setImageSrc(null)
+    setCrop({ x: 0, y: 0 })
+    setZoom(1)
+    setCroppedAreaPixels(null)
+    setUserMsg(undefined)
+
+    if (fileInputRef.current) {
+      fileInputRef.current.value = ""
+    }
+  }
+
+  const handleFileChange = async (e: React.ChangeEvent<HTMLInputElement>) => {
+    const file = e.target.files?.[0]
+
+    if (file) {
+      const reader = new FileReader()
+
+      reader.onload = () => setImageSrc(reader.result as string)
+      reader.readAsDataURL(file)
+    }
+  }
+
+  const handleCropComplete = useCallback((croppedArea: any, croppedAreaPixels: any) => {
+    setCroppedAreaPixels(croppedAreaPixels)
+  }, [])
+
+  const showErrorMsg = () => {
+    setUserMsg("Failed to update avatar. Please try again.")
+
+    setTimeout(() => {
+      setUserMsg(undefined)
+    }, msgShowDuration)
+  }
+
+  const handleCloudinaryUploadSuccess = async (url: string) => {
+    try {
+      setUserMsg("Avatar updated successfully in the cloud! Updating Nexus dawn...")
+
+      await updateUserInfo("image", url)
+      await fetchUserData("image")
+
+      setUserMsg("All updates complete! Enjoy your new avatar!")
+
+      setTimeout(() => {
+        closeModal()
+        setUserMsg(undefined)
+      }, msgShowDuration)
+    } catch (error) {
+      showErrorMsg()
+    }
+  }
+
+  const handleCrop = async () => {
+    if (!imageSrc || !croppedAreaPixels) return
+
+    setIsCropping(true)
+
+    try {
+      const croppedImage = await getCroppedImg(imageSrc, croppedAreaPixels, username)
+      const formData = new FormData()
+
+      formData.append("file", croppedImage)
+
+      const response = await customFetch(
+        "/api/profiles/upload/avatar",
+        {
+          method: "PUT",
+          body: formData,
+        },
+        true
+      )
+
+      if (response.new_url) {
+        handleCloudinaryUploadSuccess(response.new_url)
+      } else {
+        throw new Error()
```

Comment:
Do you think the error messaging should be more robust or this is enough given the context of the component? Couple thoughts that crossed my mind are bubbling up the error codes from the server or having a longer error message display time/manual close modal method? Let me know what your thoughts are.

Reply:
I think this works well enough. After the error message timeout the modal will revert to showing the cropper and its associated buttons, one of which is the cancel button. This will give the user the opportunity to try again or cancel

Reply:
This will be fine. What I meant by a manual close would be the error message hanging for the user until they dismissed it. I may not have communicated that clearly, but It's not necessary right now anyways.